### PR TITLE
feat(ci): initialize ci from other rollkit repos

### PR DESCRIPTION
**Pull request by MSevey**
## Overview

Initializing CI to bring repo up to standard

## Summary by CodeRabbit

- **New Features**
	- Introduced automated review settings for pull requests, ensuring core team members are assigned as reviewers.
	- Added automated dependency updates for GitHub Actions and Go modules, enhancing project maintainability.
	- Established CI and release workflows to streamline code quality checks and version management.
	- Implemented housekeeping tasks to automate issue and pull request management.
	- Introduced linting and testing workflows to ensure code quality and coverage.

- **Bug Fixes**
	- Enhanced linting configurations to improve code quality checks across various file types.

- **Documentation**
	- Added configuration files for linting rules, providing clear guidelines for Markdown and YAML files.

### CODE REVIEW

**Comment by MSevey** [comment]
@coderabbittai ignore

**Comment by MSevey** [comment]
@coderabbittai resolve

File: relayer.go [hunk 1]
```diff
@@ -111,7 +111,7 @@ type Relayer struct {
 }
 
 // close shuts down the client.
-func (r Relayer) close() {
+func (r Relayer) close() { // nolint:unused
```
Comment (MSevey) [inline relayer.go:114]: @tuxcanfly is this ok? should it be used somewhere

File: relayer.go [hunk 2]
```diff
@@ -203,14 +203,14 @@ func (r Relayer) revealTx(embeddedData []byte, commitHash *chainhash.Hash) (*cha
...
-		return nil, fmt.Errorf("error building p2tr script: %v", err)
+		return nil, fmt.Errorf("error building p2tr script: %w", err)
 	}
 
 	tx := wire.NewMsgTx(2)
 	tx.AddTxIn(&wire.TxIn{
 		PreviousOutPoint: wire.OutPoint{
 			Hash:  *rawCommitTx.Hash(),
-			Index: uint32(commitIndex),
+			Index: uint32(commitIndex), // nolint:gosec
```
Comment (MSevey) [inline relayer.go:213]: overflow

**Review by Manav-Aggarwal — APPROVED** [review]

**Review by tzdybal — APPROVED** [review]

**Pull request merged**